test/test_jit.py

```diff
@@ -1764,6 +1764,25 @@ def forward(self, x):
             "Expected to have 2 relu modules after dedup module uses"
         self.assertEqual(res, ref_res)
 
+    def test_duplicate_dequant(self):
+        class M(torch.nn.Module):
+            def __init__(self):
+                super(M, self).__init__()
+                self.conv = torch.nn.Conv2d(3, 3, 3)
+
+            def forward(self, x):
+                x = torch.dequantize(x)
+                r = self.conv(x)
+                r += x
+                return r
+        m = torch.jit.script(M())
+        FileCheck().check_count("aten::dequantize", 1, exactly=True) \
+                   .run(m.graph)
+        torch._C._jit_pass_replicate_dequantize(m.graph)
+        FileCheck().check_count("aten::dequantize", 2, exactly=True) \
```

Comment:
Can we do a pattern match with the expected graph to ensure that both the input to conv and add use dequantized value of x?

Reply:
FileCheck() only supports string match, I can add a test for numerics to make sure the transformation doesn't change the result